### PR TITLE
Add option to disable autoplay of video #1996

### PR DESCRIPTION
**Pull request by whitecrownclown**
Hello @night ,

First of all, great extension, twitch is a lot better with it. Please have a look over this PR and let me know if you have any feedback.

Thanks!

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,7120 @@
+{
```
Comment (night) [inline package-lock.json]: lets not commit this in this PR.

File: src/watcher.js [hunk 2]
```diff
@@ -341,7 +378,8 @@ class Watcher extends SafeEventEmitter {
             if (currentChannel.id === channel.id) return;
             channel = currentChannel;
 
-            api.get(`channels/${channel.name}`)
+            api
```
Comment (night) [inline src/watcher.js]: it looks like you have some autoformatter on that is causing unrelated changes. please disable that and undo these unrelated changes.

Comment (whitecrownclown) [reply src/watcher.js]: yeah, i've prettier autoformatting that for me, i'll revert these

File: package-lock.json (hunk 1 above)
Comment (whitecrownclown) [reply package-lock.json]: Sure, i'll remove it from the PR

File: src/watcher.js [hunk 3]
```diff
@@ -333,6 +343,33 @@ class Watcher extends SafeEventEmitter {
         this.on('load.vod', () => observe(vodChatWatcher, $('.qa-vod-chat')[0]));
     }
 
+    vodRecommendationObserver() {
+        vodRecommendationWatcher = new window.MutationObserver(mutations => {
```
Comment (night) [inline src/watcher.js]: rather than watching for vod recommendations, why not hook into the player events themselves and watch for the video end event and prevent the next video from being played?

File: src/modules/disable_vod_autoplay/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const $ = require('jquery');
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+class DisableVodAutoplay {
+    constructor() {
+        settings.add({
+            id: 'disableVodAutoplay',
+            name: 'Disable Vod Autoplay',
```
Comment (night) [inline src/modules/disable_vod_autoplay/index.js]: this setting is not self explanatory. i had to read the code to understand what "disable vod autoplay" meant. it should be something named around disabling autoplay of recommended videos

File: src/watcher.js (hunk 3 above)
Comment (whitecrownclown) [reply src/watcher.js]: would this be the appropriate event to listen to `Twitch.Player.ENDED` ?

**Comment by whitecrownclown** [comment]
@night Hi, I've gone through the changes and also followed your advice about using the player.ended event.

**Comment by whitecrownclown** [comment]
@night Can you have a look over this PR again?
Thanks!

**Comment by night** [comment]
i am OOO til next week. will look then as this is not urgent.

File: src/modules/video_player/index.js [hunk 5]
```diff
@@ -17,6 +18,22 @@ function stepPlaybackSpeed(faster) {
     currentPlayer.player.setPlaybackRate(rates[idx]);
 }
 
+function watchPlayerRecommendationVodsAutoplay() {
+    const player = twitch.getCurrentPlayer().player;
```
Comment (night) [inline src/modules/video_player/index.js]: getCurrentPlayer can return null/undefined .. you should not grab player off of it

File: src/watcher.js [hunk 6]
```diff
@@ -359,7 +360,7 @@ class Watcher extends SafeEventEmitter {
         const observe = (watcher, element) => {
             if (!element) return;
             if (watcher) watcher.disconnect();
-            watcher.observe(element, {childList: true, subtree: true});
+            watcher.observe(element, { childList: true, subtree: true });
```
Comment (night) [inline src/watcher.js]: what's with these spaces

File: src/modules/video_player/index.js [hunk 7]
```diff
@@ -74,6 +94,12 @@ class VideoPlayerModule {
             defaultValue: false,
             description: 'Click on the twitch player to pause/resume playback'
         });
+        settings.add({
+            id: 'disableVodAutoplay',
+            name: 'Disable VOD Autoplay',
```
Comment (night) [inline src/modules/video_player/index.js]: VoD

File: src/modules/video_player/index.js (hunk 5 above)
Comment (whitecrownclown) [reply src/modules/video_player/index.js]: Is there any recommended way of getting player or should i just guard for it?

**Comment by whitecrownclown** [comment]
Ready for another go @night
